wasm/main.go

```diff
@@ -0,0 +1,77 @@
+package main
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"strings"
+	"syscall/js"
+
+	"github.com/centrifuge/precise-proofs/examples/documents"
+	"github.com/centrifuge/precise-proofs/proofs"
+	"github.com/centrifuge/precise-proofs/proofs/proto"
+	"github.com/golang/protobuf/jsonpb"
+)
+
+func checkErr(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+func generateProof(i []js.Value) {
```

Comment:
How does the compiler treat and convert return statements to JS functions? In this implementation, I believe wasm will convert a `fmt` standard output to a `console.log` which is good for debugging. On the other hand, we would like to capture a JSON as the return of this function. 
Do you know how the mapping converters work and over which types?